src/store/token.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+export async function getToken(token) {
+	const tokenValue = await ((typeof (token) === 'function')
+		? token()
+		: token);
+
+	return await new Token(tokenValue);
+}
+
+export function compareTokens(a, b) {
+	return a.toString() === b.toString();
+}
+
+export const TOKEN_COOKIE = -1;
+export const TOKEN_COOKIE_CACHE_KEY = 'cookie';
+
+class Token {
+	constructor(token) {
+		this._cacheKey = this._praseCacheKey(token);
+		this._value = token;
+	}
+
+	get value() {
+		return this._value;
+	}
+
+	toString() {
+		return this._cacheKey;
+	}
+
+	isResolved() {
+		return !!this.value;
+	}
+
+	/**
+	 * From the static data from the JWT create a static key.
+	 * @param {*} token
+	 */
+	_praseCacheKey(token) {
```

Comment:
Should this be `parse` instead of `prase`?

Reply:
Yup that is a typo thank you! :)

Reply:
This method does not access any state of the class so it could be static or you could move it outside the class and just use it as a standalone, non-exported function. The latter would also make it truly private. Another option, would be to just put all this code in the constructor since that's the only place this method is called from and the constructor is not doing anything else really.

Reply:
Is this function just parsing a JWT? If so, I'm sure there is an existing npm library we could use for this.

Reply:
It is taking our token then only parsing the data that doesn't change between token refreshes. I don't think using a parser would add anything. It wouldn't replace this method.

Reply:
I agree that a parser would not functionally add anything but it's code that someone else writes/maintains/tests so less work for us. I don't currently see a test for this code so that's additional work you'll eventually need to do.

Reply:
There are tests in the other repo for this. I'll be bringing it over after I have the architecture sorted.